javalib/src/main/scala/java/lang/System.scala

```diff
@@ -61,7 +63,14 @@ object System {
         sysProps.setProperty("user.language", userLang)
         sysProps.setProperty("user.country", userCountry)
       }
-      sysProps.setProperty("user.home", getenv("HOME"))
+      {
```

Comment:
Consider using
```suggestion
      locally {
```
to make this more readable.

Also, consider wrapping the following code for `user.dir` in a similar `locally {}` block, since it also has a `val buf`.

Reply:
Did not know about `locally` existence! Thank you :-) I'll implement the suggested changes!